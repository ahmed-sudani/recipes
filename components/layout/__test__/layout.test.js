import { render, screen } from '@testing-library/react'
import Layout from '..'

describe('Layout', () => {
  it('should render layout component', () => {
    render(
      <span data-testid="layout">
        <Layout />
      </span>
    )
    const layoutElement = screen.getByTestId(/layout/i)
    expect(layoutElement).toMatchSnapshot()
  })
})
