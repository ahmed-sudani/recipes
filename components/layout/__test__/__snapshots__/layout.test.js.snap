// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should renders correctly 1`] = `
<span
  data-testid="layout"
>
  <header
    class="header"
    data-testid="header"
  >
    <div
      class="icon"
    >
      <div />
      <div>
        Intouch Recipes
      </div>
    </div>
    <input
      class="checkInput"
      id="side-menu"
      type="checkbox"
    />
    <label
      class="checkBtn"
      for="side-menu"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 16 16"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2.5 12a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5z"
          fill-rule="evenodd"
        />
      </svg>
    </label>
    <nav
      class="nav"
    >
      <div
        class="link"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 13.5V7h1v6.5a.5.5 0 0 0 .5.5h9a.5.5 0 0 0 .5-.5V7h1v6.5a1.5 1.5 0 0 1-1.5 1.5h-9A1.5 1.5 0 0 1 2 13.5zm11-11V6l-2-2V2.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5z"
            fill-rule="evenodd"
          />
          <path
            d="M7.293 1.5a1 1 0 0 1 1.414 0l6.647 6.646a.5.5 0 0 1-.708.708L8 2.207 1.354 8.854a.5.5 0 1 1-.708-.708L7.293 1.5z"
            fill-rule="evenodd"
          />
        </svg>
        <div>
          Home
        </div>
      </div>
      <div
        class="link"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 11a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
          />
          <path
            d="M13.997 5.17a5 5 0 0 0-8.101-4.09A5 5 0 0 0 1.28 9.342a5 5 0 0 0 8.336 5.109 3.5 3.5 0 0 0 5.201-4.065 3.001 3.001 0 0 0-.822-5.216zm-1-.034a1 1 0 0 0 .668.977 2.001 2.001 0 0 1 .547 3.478 1 1 0 0 0-.341 1.113 2.5 2.5 0 0 1-3.715 2.905 1 1 0 0 0-1.262.152 4 4 0 0 1-6.67-4.087 1 1 0 0 0-.2-1 4 4 0 0 1 3.693-6.61 1 1 0 0 0 .8-.2 4 4 0 0 1 6.48 3.273z"
          />
        </svg>
        <div>
          Recipes
        </div>
      </div>
      <div
        class="link"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
          />
          <path
            d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
          />
        </svg>
        <div>
          New
        </div>
      </div>
      <div
        class="link"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8 2.748-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
          />
        </svg>
        <div>
          Favorites
        </div>
      </div>
      <div
        class="link"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.478 1.647a.5.5 0 1 0-.956-.294l-4 13a.5.5 0 0 0 .956.294l4-13zM4.854 4.146a.5.5 0 0 1 0 .708L1.707 8l3.147 3.146a.5.5 0 0 1-.708.708l-3.5-3.5a.5.5 0 0 1 0-.708l3.5-3.5a.5.5 0 0 1 .708 0zm6.292 0a.5.5 0 0 0 0 .708L14.293 8l-3.147 3.146a.5.5 0 0 0 .708.708l3.5-3.5a.5.5 0 0 0 0-.708l-3.5-3.5a.5.5 0 0 0-.708 0z"
          />
        </svg>
        <div>
          Developers
        </div>
      </div>
      <button
        class="signInBtn"
      >
        Sign in
      </button>
    </nav>
  </header>
  <main />
  <footer
    class="footer"
  >
     all copy right reserved
  </footer>
</span>
`;
